agent/http.go

```diff
@@ -631,13 +652,24 @@ func (s *HTTPServer) parseMetaFilter(req *http.Request) map[string]string {
 	return nil
 }
 
-// parse is a convenience method for endpoints that need
+// parseResolveProxy is a convenience method for endpoints that need
 // to use both parseWait and parseDC.
-func (s *HTTPServer) parse(resp http.ResponseWriter, req *http.Request, dc *string, b *structs.QueryOptions) bool {
+func (s *HTTPServer) parseResolveProxy(resp http.ResponseWriter, req *http.Request, dc *string, b *structs.QueryOptions, resolveProxyToken bool) bool {
```

Comment:
Names are hard :(

I can't think of anything better but it's a bit unintuitive that parseResolveProxy optionally doesn't resolve the proxy! I'd be tempted to name it something like `parseInternal` or `parseWithOptions` to indicate that it's mostly a way to de-duplicate logic from higher level callers?

Not a blocker at all though.